### PR TITLE
update graphics with sdl2 changes

### PR DESCRIPTION
**Pull request by myk002**
the `std::variant` representation may not be the best

### CODE REVIEW

**Comment by cvuchener** [comment]
Where is the std::variant?

**Comment by myk002** [comment]
in g_src: https://github.com/Putnam3145/Dwarf-Fortress--libgraphics--/blob/sdl2/g_src/graphics.h#L2985

File: df.graphics.xml [hunk 1]
```diff
@@ -190,6 +190,19 @@
         <pointer name='edge_biome_se' type-name='int32_t' is-array='true'/>
     </struct-type>
 
+    <struct-type type-name='cached_texturest'>
+        <int32_t name='w'/>
+        <int32_t name='h'/>
+        <pointer name='tex'/>
+        <uint8_t name='tex_n'/>
```
Comment (ab9rf) [inline df.graphics.xml:197]: tag comes before the variant representation, not after

**Review by ab9rf — CHANGES_REQUESTED** [review]

File: df.graphics.xml (hunk 1 above)
Comment (ab9rf) [reply df.graphics.xml:197]: nevermind, i'm wrong, this is correct except that the tag should be `int8_t` as `-1` is always a valid value for the union tag field

**Review by ab9rf — APPROVED** [review]

**Pull request merged**